components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -429,6 +447,79 @@ public String exportApplication(String applicationId, Boolean exportSecrets) {
         }
     }
 
+    /**
+     * Export an application identified by the applicationId, in the given format.
+     *
+     * @param fileType The format of the exported string.
+     * @param applicationId ID of the application to be exported.
+     * @param exportSecrets If True, all hashed or encrypted secrets will also be exported.
+     * @return string of the application in the given format.
+     */
+    public TransferResource exportApplicationAsFile(String fileType, String applicationId, Boolean exportSecrets) {
+
+        if (StringUtils.isBlank(fileType)) {
+            throw new UnsupportedOperationException("No valid media type found");
+        }
+
+        ServiceProvider serviceProvider;
+        try {
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            serviceProvider = getApplicationManagementService().exportSPFromAppID(
+                    applicationId, exportSecrets, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            String msg = "Error exporting application with id: " + applicationId;
+            throw handleIdentityApplicationManagementException(e, msg);
+        }
+
+        return  generateFileFromModel(fileType, serviceProvider);
+    }
+
+    private TransferResource generateFileFromModel(String fileType, ServiceProvider serviceProvider) {
+
+        StringBuilder fileNameSB = new StringBuilder(serviceProvider.getApplicationName());
+        String value = "";
+        switch (fileType) {
+            case MEDIA_TYPE_APPLICATION_YAML:
+                Yaml yaml = new Yaml();
+                try {
+                    value = yaml.dump(serviceProvider);
+                } catch (NullPointerException | YAMLException | ClassCastException e) {
+                    throw new RuntimeException(e);
```

Comment:
Any reason to throw a runtime exception here?

Reply:
Changed this to throw a server error.